src/DepthCameraSensor.cc

```diff
@@ -582,6 +582,19 @@ bool DepthCameraSensor::Update(
     *this->dataPtr->pointMsg.mutable_header()->mutable_stamp() =
       msgs::Convert(_now);
     this->dataPtr->pointMsg.set_is_dense(true);
+    // Set frame_id
+    for (auto i = 0;
+         i < this->dataPtr->pointMsg.mutable_header()->data_size();
+         ++i)
+    {
+      if (this->dataPtr->pointMsg.mutable_header()->data(i).key() == "frame_id"
+          && this->dataPtr->pointMsg.mutable_header()->data(i).value_size() > 0)
+      {
+        this->dataPtr->pointMsg.mutable_header()->mutable_data(i)->set_value(
+              0,
+              this->FrameId());
+      }
+    }
```

Comment:
Can you try and see if just changing the [msgs::InitPointCloudPacked](https://github.com/gazebosim/gz-sensors/blob/734c1f28d5f03379acfae728b31d1143ddff34ed/src/DepthCameraSensor.cc#L307-L309) call to use `this->FrameId()` (instead of `this->Name()`) will work?

```cpp
  msgs::InitPointCloudPacked(this->dataPtr->pointMsg, this->FrameId(), true,
      {{"xyz", msgs::PointCloudPacked::Field::FLOAT32},
       {"rgb", msgs::PointCloudPacked::Field::FLOAT32}});
```

That'll simplify the logic and remove the need for the for loop.

Reply:
Works like a charm! Will be amending to this commit.